src/codegen/__future__/definePlugin.ts

```diff
@@ -0,0 +1,211 @@
+import { OpenAPIV3 } from "openapi-types";
+import { type SchemaObject } from "../generate";
+import ts from "typescript";
+import { Opts } from "../index";
+import minimist from "minimist";
+import assert from "node:assert";
+
+export type OazapftsCodegenPluginDefinition = {
+  /**
+   * Required identifier for this plugin
+   */
+  name: string;
```

Comment:
I'd suggest to use the plugin object itself as an identifier. 
The `name` would still be useful for logging when needed. But might not be required.

Reply:
I've made this not required